UnitTest/tests/test_SDUT-devices.txt

```diff
@@ -0,0 +1,200 @@
+																	"rmsg"         => "MU;P0=-2009;P1=2076;P2=-259;P3=329;P4=-15751;P5=1216;D=012121212303012303034121212301212303012121212303012303034121212301212303012121212303012303034121212305;CP=3;R=37;",
+																	"state"        => "Button_1",
+																	"tristateCode" => "1F10110F0",
+																	"model"        => "Tedsen_SKX2xx"
+																},
+		"Tedsen_SKX6xx_1F10FF0" =>  {	"DEF"          => "Tedsen_SKX6xx 1F10FF0",
+																	"rmsg"         => "MU;P0=-32001;P1=2072;P2=-260;P3=326;P4=-2015;P5=-15769;D=01212123412123434123412343434123412151212123412123434123412343434123412151212123412123434123412343434123412151212123412123434123412343434123412151212123412123434123412343434123412151212123412123434123412343434123412151212123412123434123412343434123412151;CP=3;R=37;O;",
+																	"state"        => "Button_6",
+																	"tristateCode" => "1F10FF0F1",
+																	"model"        => "Tedsen_SKX6xx"
+																},
+		"Unitec_47031_95" =>  {	"DEF"   => "Unitec_47031 95",
+														"rmsg"  => "MU;P0=309;P1=636;P2=-690;P3=-363;P4=-10027;D=012031203120402031312031203120312031204020313120312031203120312040203131203120312031203120402031312031203120312031204020313120312031203120312040203131203120312031203120402031312031203120312031204020313120312031203120312040203131203120312030;CP=0;O;",
+														"state" => "1",
+														"Zone"  => "010 (off|on|off) - Zone 3",
+														"ToDo"  => "added to JSON and recheck",
+														"model" => "Unitec_47031"
+													},
+	);
+	
+	for my $dev (sort keys %devices) {
+		subtest "Checking $dev" => sub {
+			my $plan = 0;
+
+			for my $key (keys %{$devices{$dev}}) {
+				#note($key);
+				$plan++ if ($key ne "DEF" && $key ne "rmsg" && $key ne "ToDo" && $key ne "model");
+			}
+			
+			plan => $plan;
+
+			if(defined($defs{$dev}) && $defs{$dev}{TYPE} eq "SD_UT" && $defs{$dev}{NAME} eq $dev) {
+				note("device is already defined on system");
+				CommandSetReading(undef, "$dev state missing");
+				#CommandDelete(undef,$dev);
+			} else {
+				note("device must defined");
+				CommandDefMod(undef,"-temporary $dev SD_UT ".$devices{$dev}{DEF});			
+			}
+
+			my %signal_parts = SIGNALduino_Split_Message($devices{$dev}{rmsg},$targetHash->{NAME});
+			SIGNALduino_Parse($targetHash, $targetHash, $targetHash->{NAME}, "\002".$devices{$dev}{rmsg}."\003");
+			
+			## all readings, if ne DEF,rmsg,ToDo ##
+			for my $key (keys %{$devices{$dev}}) {
+				#note($key);
+				is(ReadingsVal($dev ,$key,"0"),$devices{$dev}{$key},"check reading $key") if ($key ne "DEF" && $key ne "rmsg" && $key ne "ToDo" && $key ne "model");
+			}
+
+			TODO: {
+				# read from internal STATE, if FHEM restart -> defined, after dispatch must changed !!
+				is(InternalVal($dev ,"STATE","0"),$devices{$dev}{state},"check internal state");
+			}
+
+			note("--> ToDo: ".$devices{$dev}{ToDo}." <--") if ($devices{$dev}{ToDo});
+			sleep 3;
```

Comment:
Brauchst Du den sleep?
Wenn nicht, spart es jede Menge Zeit :)

Reply:
Den sleep habe ich reingenommen weil du ihn auch überall drin hast :-D
Copy und Past aber ich werde es mal testen morgen und ohne laufen lassen. Zeit, die spielt keine Rolle.

PS: Dein Modul solltest du als nonBlocking schreiben :-P Denn wenn man das lokal testet am System, blockiert man bei den Tests das sämtliche System ;-)

Reply:
Ich hatte mal geforkt.
Das kannst Du auch über ein Attribut umstellen.

Mit geforkten Prozessen funktioniert devel::Cover nicht und daher bin ich wieder auf die blockierende Variante umgestiegen.

Ich nehme an, dass man das Modul ohnehin nur in einer Test-Instanz verwendet.

Reply:
Nö, ich habe es hier lokal weil ich sehen will was als Ergebnis herauskommt :-)
Die Erarbeitung am livesystem oder Clonesystem ist direkter finde ich.